hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java

```diff
@@ -2265,14 +2264,9 @@ public void onConfigurationChange(Configuration newConf) {
     if (scheduler instanceof ConfigurationObserver) {
       ((ConfigurationObserver)scheduler).onConfigurationChange(newConf);
     }
-    // Make sure authManager will read hbase-policy file
-    System.setProperty("hadoop.policy.file", "hbase-policy.xml");
-    synchronized (authManager) {
-      authManager.refresh(newConf, new HBasePolicyProvider());
+    if (authorize) {
```

Comment:
btw, this if condition is an enhancement for this patch? is it necessary?

Reply:
Yes, it is, to avoid unnecessary refresh. If authorization is disabled, then no need to refresh it which it is time consuming.

We should forward-port this patch to branch-2+ and master as well, quite some changes.

Reply:
👍